internal/adminapi/backoff_strategy_konnect.go

```diff
@@ -0,0 +1,131 @@
+package adminapi
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jpillora/backoff"
+	"github.com/kong/go-kong/kong"
+	"github.com/samber/lo"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/deckerrors"
+)
+
+const (
+	KonnectBackoffInitialInterval = time.Second * 3
+	KonnectBackoffMaxInterval     = time.Minute * 15
+	KonnectBackoffMultiplier      = 2
+)
+
+type Clock interface {
+	Now() time.Time
+}
+
+type SystemClock struct{}
+
+func (SystemClock) Now() time.Time { return time.Now() }
+
+// KonnectBackoffStrategy keeps track of Konnect config push backoffs.
+type KonnectBackoffStrategy struct {
```

Comment:
I'm missing why this is specific to Konnect. Why couldn't this be just a generic implementation of backoff strategy? 🤔

Reply:
Yes, it could be used for regular gateways as well, but by naming it this way I wanted to emphasize that it's meant to be used specifically with the Konnect client as of now. It's because we have no clear decision whether we'd like to follow the very same approach with regular gateways.

@mflendrich had a point about [400-500) code in case of a regular gateway may not be always considered non-recoverable without a config change, because there's a possibility someone has part of their configuration managed by KIC, and the other directly via deck/admin API which potentially can result in KIC not being able to push some entity due to unique constraints, e.g. entity name used in the config part managed manually is a duplicate of the name generated by KIC. That's a case where we cannot safely assume KIC cannot recover without config change since if a user deletes the manually managed part of the config, KIC will be able to push its part eventually. 

In the case of Konnect, we don't have to worry about this since KIC is the only one in charge of the configuration on Konnect side.

Reply:
In such case this will only work for read-only integration with Konnect which we currently support. If we ever move to read write this will break, correct?

Reply:
Yes, there's such a possibility, although I'm not aware of any plans that will make this happen in the nearest future. I left a comment though on the strategy to mention that for the future us.